src/modules/commander/Commander.cpp

```diff
@@ -2828,35 +2865,15 @@ Commander::run()
 
 		// automatically set or update home position
 		if (_param_com_home_en.get() && !_home_pub.get().manual_home) {
-			const vehicle_local_position_s &local_position = _local_position_sub.get();
-
-			if (!_armed.armed) {
-				if (_home_pub.get().valid_lpos) {
-					if (_vehicle_land_detected.landed && local_position.xy_valid && local_position.z_valid) {
-						/* distance from home */
-						float home_dist_xy = -1.0f;
-						float home_dist_z = -1.0f;
-						mavlink_wpm_distance_to_point_local(_home_pub.get().x, _home_pub.get().y, _home_pub.get().z,
-										    local_position.x, local_position.y, local_position.z,
-										    &home_dist_xy, &home_dist_z);
-
-						if ((home_dist_xy > local_position.eph * 2.0f) || (home_dist_z > local_position.epv * 2.0f)) {
-
-							/* update when disarmed, landed and moved away from current home position */
-							set_home_position();
-						}
-					}
-
-				} else {
-					/* First time home position update - but only if disarmed */
+			if (!_armed.armed && _vehicle_land_detected.landed) {
+				const bool can_set_home_lpos_first_time = (!_home_pub.get().valid_lpos && _status_flags.local_position_valid);
+				const bool can_set_home_gpos_first_time = ((!_home_pub.get().valid_hpos || !_home_pub.get().valid_alt)
+						&& (_status_flags.global_position_valid || _status_flags.gps_position_valid));
+
+				if (can_set_home_lpos_first_time
+				    || can_set_home_gpos_first_time
+				    || hasMovedFromCurrentHomeLocation()) {
```

Comment:
So setting home upon takeoff is not a desired thing? I'm just asking because I don't know and this pr also states uncertainty in the description: https://github.com/PX4/PX4-Autopilot/pull/14585

Reply:
Yes, we want to set home on takeoff, this is handled here: https://github.com/PX4/PX4-Autopilot/pull/19437/files#diff-e4b611e75eba3844661619867ee06befbcf230aced354a14f167f7d0ebbca8e0R2226